packages/mobile/src/verify/VerificationInputScreen.tsx

```diff
@@ -2,93 +2,229 @@ import Link from '@celo/react-components/components/Link'
 import TextButton from '@celo/react-components/components/TextButton'
 import TextInput from '@celo/react-components/components/TextInput'
 import withTextInputPasteAware from '@celo/react-components/components/WithTextInputPasteAware'
+import Checkmark from '@celo/react-components/icons/Checkmark'
 import SmsCeloSwap from '@celo/react-components/icons/SmsCeloSwap'
 import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
+import { stripHexLeader } from '@celo/utils/src/signatureUtils'
 import { extractAttestationCodeFromMessage } from '@celo/walletkit'
 import dotProp from 'dot-prop-immutable'
+import { padStart } from 'lodash'
 import * as React from 'react'
 import { withNamespaces, WithNamespaces } from 'react-i18next'
-import { ScrollView, StyleSheet, Text, View } from 'react-native'
+import { ActivityIndicator, BackHandler, ScrollView, StyleSheet, Text, View } from 'react-native'
 import Modal from 'react-native-modal'
 import SafeAreaView from 'react-native-safe-area-view'
 import { connect } from 'react-redux'
-import { hideAlert, showError } from 'src/alert/actions'
+import { hideAlert } from 'src/alert/actions'
 import { errorSelector } from 'src/alert/reducer'
 import componentWithAnalytics from 'src/analytics/wrapper'
 import { ErrorMessages } from 'src/app/ErrorMessages'
+import CancelButton from 'src/components/CancelButton'
 import DevSkipButton from 'src/components/DevSkipButton'
-import { Namespaces } from 'src/i18n'
+import i18n, { Namespaces } from 'src/i18n'
 import LoadingSpinner from 'src/icons/LoadingSpinner'
+import { cancelVerification, receiveAttestationMessage } from 'src/identity/actions'
+import { ATTESTATION_CODE_PLACEHOLDER } from 'src/identity/reducer'
 import {
-  cancelVerification,
-  receiveAttestationMessage,
-  startVerification,
-} from 'src/identity/actions'
-import { AttestationCode } from 'src/identity/verification'
+  AttestationCode,
+  CodeInputType,
+  NUM_ATTESTATIONS_REQUIRED,
+  VerificationStatus,
+} from 'src/identity/verification'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { RootState } from 'src/redux/reducers'
-import { currentAccountSelector } from 'src/web3/selectors'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'VerificationInputScreen'
 
 const CodeInput = withTextInputPasteAware(TextInput)
 
+function getRecodedAttestationValue(attestationCode: AttestationCode, t: i18n.TranslationFunction) {
```

Comment:
Actually Recoded is what I meant here, as in recoded into base64.
Also FYI isn't new code, it's from the old Verifying.tsx file, though I'm open to moving it to a new file. I kept it here cause it's kind of a hack specifically for this screen to show the same code in the UI as the user sees in their SMS

Reply:
ohhhhh. gotcha gotcha. not important but maybe then just `getBase64AttestationValue` or even just `getDisplayAttestationCode` since it also might return a string with code accepted